core/Azure.Mcp.Core/src/Services/Http/HttpClientFactoryConfigurator.cs

```diff
@@ -0,0 +1,151 @@
+        // within the service collection.
+        services.ConfigureHttpClientDefaults(static builder => ConfigureHttpClientBuilder(builder));
+
+        return services;
+    }
+
+    private static void ConfigureHttpClientBuilder(IHttpClientBuilder builder)
+    {
+        builder.ConfigureHttpClient((serviceProvider, client) =>
+        {
+            var httpClientOptions = serviceProvider.GetRequiredService<IOptions<HttpClientOptions>>().Value;
+            client.Timeout = httpClientOptions.DefaultTimeout;
+
+            var transport = serviceProvider.GetRequiredService<IOptions<ServiceStartOptions>>().Value.Transport;
+            client.DefaultRequestHeaders.UserAgent.ParseAdd(BuildUserAgent(transport));
+        });
+
+        builder.ConfigurePrimaryHttpMessageHandler(serviceProvider =>
+        {
+            var httpClientOptions = serviceProvider.GetRequiredService<IOptions<HttpClientOptions>>().Value;
+            return CreateHttpMessageHandler(httpClientOptions);
+        });
+    }
+
+    private static HttpMessageHandler CreateHttpMessageHandler(HttpClientOptions options)
+    {
+        var handler = new HttpClientHandler();
+
+        var proxy = CreateProxy(options);
+        if (proxy != null)
+        {
+            handler.Proxy = proxy;
+            handler.UseProxy = true;
+        }
+
+#if DEBUG
+        var testProxyUrl = Environment.GetEnvironmentVariable("TEST_PROXY_URL");
+        if (!string.IsNullOrWhiteSpace(testProxyUrl) && Uri.TryCreate(testProxyUrl, UriKind.Absolute, out var proxyUri))
+        {
+            return new RecordingRedirectHandler(proxyUri)
+            {
+                InnerHandler = handler
+            };
+        }
+#endif
+
+        return handler;
+    }
+
+    private static WebProxy? CreateProxy(HttpClientOptions options)
+    {
+        string? proxyAddress = options.AllProxy ?? options.HttpsProxy ?? options.HttpProxy;
+
+        if (string.IsNullOrEmpty(proxyAddress))
+        {
+            return null;
+        }
+
+        if (!Uri.TryCreate(proxyAddress, UriKind.Absolute, out var proxyUri))
+        {
+            return null;
+        }
+
+        var proxy = new WebProxy(proxyUri);
+
+        if (!string.IsNullOrEmpty(options.NoProxy))
+        {
+            var bypassList = options.NoProxy
+                .Split(',', StringSplitOptions.RemoveEmptyEntries)
+                .Select(s => s.Trim())
+                .Where(s => !string.IsNullOrEmpty(s))
+                .Select(ConvertGlobToRegex)
+                .ToArray();
+
+            if (bypassList.Length > 0)
+            {
+                proxy.BypassList = bypassList;
+            }
+        }
+
+        return proxy;
+    }
+
+    private static string ConvertGlobToRegex(string globPattern)
+    {
+        if (string.IsNullOrEmpty(globPattern))
+        {
+            return string.Empty;
+        }
+
+        var escaped = globPattern
```

Comment:
Future PR suggestion: 

Calling `string.Replace` creates a new `string` each time because `string` contents cannot be modified. It'd be more efficient if we could use a sort of `ReadOnlySpan<char>.Replace` method (or extension method) to just use stack-allocated memory, but it looks to be an [open issue for .NET.](https://github.com/dotnet/runtime/issues/29758)

Recreating the regex is fine if we want to support changes to `HttpClientOptions` during the runtime of the application, which would require some amount of recalculating (or at least a simple cache) each `HttpClient`/`HttpMessageHandler` creation.

I'm calling this "future PR suggestion" because I don't know if we expect this to run much in the wild or only for our test automation.